src/get-variables-functions.ts

```diff
@@ -31,12 +31,12 @@ export const getStateVariables = (contractNode: ContractDefinitionNode): StateVa
     // Get the type of the state variable
     const stateVariableType: string = stateVariableNode.typeDescriptions.typeString;
 
-    // If nested mapping return
-    if (stateVariableType.includes('=> mapping')) return;
-
     // Check if the state variable is an array or a mapping or a basic type
     if (stateVariableType.startsWith('mapping')) {
+      // If nested mapping return
+      if (stateVariableType.includes('=> mapping')) return;
       // If value is of type struct we don't mock it
+      if (stateVariableType.includes('=> struct')) return;
```

Comment:
Regarding the solution, the current setup retrieves data through the `ast` object, which doesn't contain detailed struct info as the `abi` object does — so mappings being nested by a struct cannot be checked.

This issue may be the reason why [BES-128](https://linear.app/defi-wonderland/issue/BES-128/look-for-alternatives-to-ast-for-parsing-contracts) exists.